### PR TITLE
providing additional information about the Consul agent

### PR DESCRIPTION
**Pull request by trujillo-adam**
One of the issues I had as someone brand new to Consul was that I didn't have a level of guidance between the getting started tutorials and the documentation. This PR is intended to cover the gap and provide a starting point for users at a similar level. It answers many of the questions I had when I first started that I couldn't figure out from the tutorials or current docs.

### CODE REVIEW

File: website/content/docs/agent/index.mdx [hunk 1]
```diff
@@ -9,33 +9,66 @@ description: >-
...
+The `-dev` flag is provided for learning purposes only.
+We strongly advise against using it for production environments.
+
+-> **Getting Started Tutorials**: You can test a local agent by following the
 [Getting Started tutorials](https://learn.hashicorp.com/tutorials/consul/get-started-install?utm_source=consul.io&utm_medium=docs).
 
-The agent command takes a variety of
-[`configuration options`](/docs/agent/options#command-line-options), but most
-have reasonable defaults.
+The only information Consul needs to run is the location of a directory for storing agent state data, specified with the `-data-dir` flag.
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/index.mdx:51]:
```suggestion
Assuming Consul in not running n `-dev` mode. The only information Consul needs to run is the location of a directory for storing agent state data, specified with the `-data-dir` flag.
```
The jump from dev to non-dev mode felt a bit abrupt.

File: website/content/docs/agent/index.mdx [hunk 2]
```diff
@@ -9,33 +9,66 @@ description: >-
...
+Specifying the data directory and no other options will start a client agent, but the agent will be unable to perform any operations or connect to any other nodes.
+
+The following example starts a Consul agent in dev mode that will store agent state data in the `tmp/consul` directory relative to the current directory:
+
+```shell-session
+consul agent -config-file=client.hcl -dev
+```
+
+In the following example, the agent configuration file would contain the following setting:
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/index.mdx:60]: This part confused me. Is the intent to showcase the ability to specify another directory for the data location?

File: website/content/docs/agent/index.mdx [hunk 3]
```diff
@@ -92,6 +116,218 @@ When running under `systemd` on Linux, Consul notifies systemd by sending
...
+
+### Server Node in a Service Mesh
+
+The following example configuration is for a server agent named "`consul-server`". The server is [bootstrapped](/docs/agent/options#_bootstrap) and the Consul GUI is enabled.
+The reason this server agent is configured for a service mesh is that the `connect` configuration is enabled. Connect is Consul's service mesh component that provides service-to-service connection authorization and encryption using mutual Transport Layer Security (TLS). Applications can use sidecar proxies in a service mesh configuration to establish TLS connections for inbound and outbound connections without being aware of Connect at all. See [Connect](/docs/connect) for details.
+
+<Tabs>
+<Tab heading="HCL">
+
+```hcl
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/index.mdx:160]: I hate to be this guy, but can we run a `terraform fmt` on this file.  That will clean up the formatting and align the values correctly. Again, sorry for being that guy 😞

File: website/content/docs/agent/index.mdx [hunk 4]
```diff
@@ -92,6 +116,218 @@ When running under `systemd` on Linux, Consul notifies systemd by sending
...
+  "connect": {
+    "enabled": true
+  }
+}
+```
+
+</Tab>
+</Tabs>
+
+### Client Node Registering a Service
```
Comment (karl-cardenas-coding) [inline website/content/docs/agent/index.mdx:206]: The order feels a bit odd. We went from server back to client context. And then after this section we jump back to server context. Maybe we should move this after we wrap up the server discussion ?

**Review by karl-cardenas-coding — COMMENTED** [review]
I really like the changes you have proposed here. It's much clearer how to begin. I left some suggestions and comments.

File: website/content/docs/agent/index.mdx (hunk 2 above)
Comment (trujillo-adam) [reply website/content/docs/agent/index.mdx:60]: Yes, I was trying to express that the agent can be configured in an external file. Is it confusing at this point? Should we just link to the section that talks about config files?

**Review by karl-cardenas-coding — APPROVED** [review]
LGTM

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/461995.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 887a23deb1baf80930c660263dc91c865f0cbe2b onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/461995)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 887a23deb1baf80930c660263dc91c865f0cbe2b onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/461995)

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/465114.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 887a23deb1baf80930c660263dc91c865f0cbe2b onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/465114)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 887a23deb1baf80930c660263dc91c865f0cbe2b onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/465114)

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/465350.

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 887a23deb1baf80930c660263dc91c865f0cbe2b onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/465436)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 887a23deb1baf80930c660263dc91c865f0cbe2b onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/465436)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 887a23deb1baf80930c660263dc91c865f0cbe2b onto `stable-website` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/465513)

**Comment by hc-github-team-consul-core** [comment]
:cherries::x: Cherry pick of commit 887a23deb1baf80930c660263dc91c865f0cbe2b onto `release/1.10.x` failed! [Build Log](https://circleci.com/gh/hashicorp/consul/465513)

**Pull request merged**